File: src/main/java/testingil/webinar/tddapi/s9/Navigator.java
package testingil.webinar.tddapi.s9;

public class Navigator {

	private DistanceProvider distProvider;

	public Navigator(Location loc, DistanceProvider distProvider) {
		this.distProvider = distProvider;
	}

	public Distance getDistanceFromDestination()	{
		Distance distance = new Distance();
		distance.value = 0;
		return distance;
	}

	public void setDestination(Location dest) {
	}

}

File: C#/TddApi/NavigationApp/s09.refactor_test_first/Navigator.cs
using System;

namespace NavigationApp.s09.refactor_test_first
{
    public class Navigator
    {
        DistanceProvider distProvider;
        public Navigator(Location loc, DistanceProvider distService)
        {
            distProvider = distService;
        }

        public Distance getDistanceFromDestination()
        {
            Distance distance = new Distance();
            distance.value = 0;
            return distance;
        }

        public void setDestination(Location dest)
        {
            
        }
    }
}